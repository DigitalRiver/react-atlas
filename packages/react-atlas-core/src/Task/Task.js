import React from "react";
import PropTypes from "prop-types";
import cx from "classnames";

class Task extends React.PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    const { className, style, title, selected, icon } = this.props;

    const taskLinkClasses = cx({
      taskLink: true,
      selected: selected
    });

    return (
      <li styleName={"task"} style={style} className={cx(className)}>
        <span styleName={taskLinkClasses}>
          {icon ? <i className={cx(icon, "ra_Task__icon-left")} /> : null}
          {title}
        </span>
      </li>
    );
  }
}

Task.propTypes = {
  /** An Object, array, or string of CSS classes to apply to CheckboxGroup.*/
  className: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
    PropTypes.array
  ]),
  /* Pass inline styling here. */
  style: PropTypes.object,
  /**
   * Text for task
   * @examples 'Item One'
   */
  title: PropTypes.string,
  /**
   * Adds selected class if true
   */
  selected: PropTypes.bool,
  /**
   * Adds font-awesome icon to left of title
   */
  icon: PropTypes.string
};

Task.defaultProps = {
  title: "Item One"
};

export default Task;
