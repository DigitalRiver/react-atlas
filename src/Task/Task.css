@import '../styles/variables.css';
.task {
	display: inline-block;
	text-align: center;
	position: relative;
}

.taskLink {
	composes: default-font from '../styles.css';
	composes: dark-gray from '../styles.css';
  	font-size: 19px;
	line-height: 27px;
	text-decoration:none;
	padding-right: 45px;
	padding-left: 45px;
	padding-top: 11.5px;
	padding-bottom: 11.5px;
	display: inline-block;
	cursor: pointer;
	text-align: center;
	position: relative;
}

.taskLink:hover {
	color: var(--primary);
}

.selected {
	composes: primary from '../styles.css';
	cursor: default;
}

.borderTop {
	composes: taskbar-border-top from '../styles.css';
}

.icon-left {
  	margin-right: 0.25rem;
  	vertical-align: middle;
}