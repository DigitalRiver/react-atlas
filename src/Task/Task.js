import React from "react";
import PropTypes from "prop-types";
import cx from "classnames";
import CSSModules from "react-css-modules";
import styles from "./Task.css";

export class Task extends React.PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    const {
      className,
      icon,
      id,
      index,
      onClick,
      selected,
      style,
      title
    } = this.props;

    const taskLinkClasses = cx({
      "taskLink": true,
      "selected": selected
    });

    return (
      <span
        id={id}
        style={style}
        className={cx(className)}
        styleName={taskLinkClasses}
        onClick={e => {
          if (onClick) {
            onClick(e, id, index);
          }
        }}
      >
        {icon ? <i styleName="icon-left" className={icon} /> : null}
        {title}
      </span>
    );
  }
}

Task.propTypes = {
  /** An Object, array, or string of CSS classes to apply to CheckboxGroup.*/
  "className": PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
    PropTypes.array
  ]),
  /**
   * Adds font-awesome icon to left of title
   */
  "icon": PropTypes.string,
  /**
   * Will set the html "id" property on the Button.
   */
  "id": PropTypes.string,
  /**
   * The index of the Task within the parent Taskbar.
   */
  "index": PropTypes.number,
  /**
   * Function that will be executed on click.
   */
  "onClick": PropTypes.func,
  /**
   * Adds selected class if true
   */
  "selected": PropTypes.bool,
  /** Pass inline styling here. */
  "style": PropTypes.object,
  /**
   * Text for task
   * @examples 'Item One'
   */
  "title": PropTypes.string
};

Task.defaultProps = {
  "title": "Item One"
};

export default CSSModules(Task, styles, { "allowMultiple": true });
