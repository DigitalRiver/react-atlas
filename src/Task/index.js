import Task from "./Task.js";
export { Task };
