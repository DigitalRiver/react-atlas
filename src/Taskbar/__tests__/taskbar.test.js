import React from "react";
import { mount } from "enzyme";
import { Taskbar } from "../index";
import { Task } from "../../Task/index";
import renderer from "react-test-renderer";

describe("Test Taskbar component", () => {
  it("Test render correctly", () => {
    const comp = (
      <Taskbar>
        <Task icon={"fa fa-id-card"} title="Item One" selected />
        <Task title="Item Two" />
        <Task title="Item Three" />
      </Taskbar>
    );
    const tree = renderer.create(comp).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("centers", () => {
    const taskbar = mount(
      <Taskbar center>
        <Task icon={"fa fa-id-card"} title="Item One" selected />
        <Task title="Item Two" />
        <Task title="Item Three" />
      </Taskbar>
    );

    expect(taskbar.find(".taskbarContainer.center").exists()).toBe(true);
  });

  it("applies className", () => {
    const taskbar = mount(
      <Taskbar className="myClass">
        <Task icon={"fa fa-id-card"} title="Item One" selected />
        <Task title="Item Two" />
        <Task title="Item Three" />
      </Taskbar>
    );

    expect(taskbar.find(".myClass.taskbar").exists()).toBe(true);
  });

  it("applies style", () => {
    const taskbar = mount(
      <Taskbar className="myClass" style={{ color: "red" }}>
        <Task icon={"fa fa-id-card"} title="Item One" selected />
        <Task title="Item Two" />
        <Task title="Item Three" />
      </Taskbar>
    );

    expect(taskbar.find(".myClass.taskbar").props().style).toEqual({
      color: "red"
    });
  });

  it("passes onClick to children", () => {
    const handleClick = jest.fn();
    const taskbar = mount(
      <Taskbar onClick={handleClick}>
        <Task icon={"fa fa-id-card"} title="Item One" selected />
        <Task title="Item Two" />
        <Task title="Item Three" />
      </Taskbar>
    );

    taskbar
      .find("Task")
      .first()
      .simulate("click");
    expect(handleClick).toBeCalled();
  });
});
