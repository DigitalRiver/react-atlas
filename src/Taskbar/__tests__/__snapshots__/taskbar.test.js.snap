// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Taskbar component Test render correctly 1`] = `
<div
  className="taskbar"
  style={undefined}
>
  <div
    className="taskbarContainer"
  >
    <span
      className="taskLink selected"
      id={undefined}
      onClick={[Function]}
      style={undefined}
    >
      <i
        className="fa fa-id-card icon-left"
      />
      Item One
    </span>
    <span
      className="taskLink"
      id={undefined}
      onClick={[Function]}
      style={undefined}
    >
      Item Two
    </span>
    <span
      className="taskLink"
      id={undefined}
      onClick={[Function]}
      style={undefined}
    >
      Item Three
    </span>
  </div>
</div>
`;
